Делегаты и события

De-Len opened the pull request.

De-Len committed:
Program.cs

Hamjoshua commented on Program.cs:
```diff
@@ -0,0 +1,852 @@
+        Console.WriteLine("5. Найти обратную матрицу");
+
+        int choice;
+        if (int.TryParse(Console.ReadLine(), out choice) && choice >= 1 && choice <= 5)
```
В 843 строке завелся монстр
Ты не должен бояться того, что choice выйдет за рамки. Программа должна просто вывести "хэндлеры не вывезли движуху, ваш choice гав-"
Ну ты понял

Hamjoshua replied:
Похвально уводить хэндлеры в сторону, когда им в лицо летит запрос гав-
Но при расширении тебе придется ручками настраивать условие для choice
О том, когда стоит прервать цепочку передачи запроса, читай на стр 249 EASY книги по паттернам

Hamjoshua replied:
И у тебя кстати уже все прописано: в switch стоит default. Если попадет нелепое значение, он переведет на default

De-Len replied:
Да, точняк)

Hamjoshua reviewed (changes requested):
Ревью

De-Len committed:
Update Program.cs

Hamjoshua commented on Program.cs:
```diff
@@ -0,0 +1,852 @@
+        }
+        else
+        {
+            Console.WriteLine("Буквы запрещены.");
```
Досадно, когда из-за одной ошибки ты не можешь произвести вычисления. Сделай меню в цикле while

Hamjoshua commented on Program.cs:
```diff
@@ -0,0 +1,852 @@
+
+    public void Execute(int choice, SquareMatrix matrix1, SquareMatrix matrix2)
+    {
+        IEvent ev = null;
```
Сокращенные переменные не есть гуд. Если будет жаловаться на "event", пиши "@event"

Hamjoshua commented on Program.cs:
```diff
@@ -0,0 +1,852 @@
+                    {
+                        throw new Exception("Размер должен быть больше 0");
+                    }
+                    catch (Exception ex)
```
Ну и этот прикол поправить надо

De-Len committed:
Update Program.cs

Fixed naming of variable "ev" to "@event"
Updated user input
Fixed exception handling
